### PR TITLE
Update Gradle to 8.11.1

### PR DESCRIPTION
**Pull request by reta**
### Description
Update Gradle to 8.11.1

### Related Issues
N/A

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [ ] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/ml-commons/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@dhrubo-os  Please add 2.x backport label, thank you!

**Comment by dhrubo-os** [comment]
> @dhrubo-os Please add 2.x backport label, thank you!

Added.

**Comment by dhrubo-os** [comment]
The last test suite is failing due to:

```
Error: Unable to download artifact(s): Artifact not found for name: ml-plugin-linux-21
        Please ensure that your artifact is not expired and the artifact was uploaded using a compatible version of toolkit/upload-artifact.
        For more information, visit the GitHub Artifacts FAQ: https://github.com/actions/toolkit/blob/main/packages/artifact/docs/faq.md

```

@reta any suggestion?

**Comment by reta** [comment]
> @reta any suggestion?

Interesting, I will take a look shortly, thanks @dhrubo-os

**Comment by reta** [comment]
@dhrubo-os seems like an intermittent issue :(

**Comment by dhrubo-os** [comment]
> @dhrubo-os seems like an intermittent issue :(

This doesn't seem like an intermittent issue, as every time I run the test, it kept failing with:

```
Error: Unable to download artifact(s): Artifact not found for name: ml-plugin-linux-21
        Please ensure that your artifact is not expired and the artifact was uploaded using a compatible version of toolkit/upload-artifact.
        For more information, visit the GitHub Artifacts FAQ: https://github.com/actions/toolkit/blob/main/packages/artifact/docs/faq.md
##[debug]Node Action run completed with exit code 1
##[debug]Finishing: Run actions/download-artifact@v4.1.7
```

@peterzhuamazon do you have any idea on this?

File: .github/workflows/CI-workflow.yml [hunk 1]
```diff
@@ -82,7 +82,7 @@ jobs:
           flags: ml-commons
           token: ${{ secrets.CODECOV_TOKEN }}
 
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
```
Comment (dhrubo-os) [inline .github/workflows/CI-workflow.yml:85]: I just merged this PR: https://github.com/opensearch-project/ml-commons/pull/3162

**Comment by dhrubo-os** [comment]
@reta @peterzhuamazon after upgrading upload artifact to v4, now this workflow is failing due to :

```

/usr/bin/docker exec  21c52b88320a9f9856ba0051c16d01ee574c33177d6ec96f5923a0df05061395 sh -c "cat /etc/*release | grep ^ID"
/__e/node20/bin/node: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by /__e/node20/bin/node)
```

I ran the workflow few times and same failure.

**Comment by peterzhuamazon** [comment]
Maybe the workflows needs to be updated to a different method until next year that we deprecate AL2.

**Comment by dhrubo-os** [comment]
> Maybe the workflows needs to be updated to a different method until next year that we deprecate AL2.

But now our linux Integ tests are failing. Should I downgrade to V3 then?

**Comment by peterzhuamazon** [comment]
Try this
https://github.com/opensearch-project/sql/blob/3b8661201efe893cd80da5f8dbd3b21333a8eda9/.github/workflows/sql-test-and-build-workflow.yml#L32-L33

https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/

**Comment by reta** [comment]
@dhrubo-os could you please take a look? thanks!

**Comment by peterzhuamazon** [comment]
> Try this https://github.com/opensearch-project/sql/blob/3b8661201efe893cd80da5f8dbd3b21333a8eda9/.github/workflows/sql-test-and-build-workflow.yml#L32-L33
> 
> https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/

Resolved in https://github.com/opensearch-project/ml-commons/pull/3223

**Comment by reta** [comment]
> Resolved in #3223

Thanks @peterzhuamazon  I will try this shortly, thanks!

**Comment by reta** [comment]
@dhrubo-os apologies, need your approval one more time

**Review by dhrubo-os — APPROVED** [review]

**Comment by reta** [comment]
@dhrubo-os good to go? ;-)

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the contribution !

**Comment by dhrubo-os** [comment]
@reta Backport PR is failing. Could you please take a look?

**Pull request merged**